### DISCUSSION TITLE
Find free-tier (or/and paid) deployment options that allow P2P

### DISCUSSION THREAD

**Discussion by KiraLT** [KiraLT/torrent-stream-server#248] [Ideas]
As discussed in #172, we need a reliable free-tier deployment service (or recommended paid service) that allows peer-to-peer.
Heroku and Octeko are great services, but they block p2p services #32.

List of providers which has a free tier: https://free-for.dev/#/?id=paas.

### Summary

- [x] ~~[Deta](https://www.deta.sh) - has one button deployment, however, has strict limitations, so streaming doesn't work: #247~~
- [ ] [fly.io](https://fly.io)
- [ ]  [Divio](https://www.divio.com)
- [ ] [Koyeb](https://www.koyeb.com)
- [ ] [Gigalixir](https://gigalixir.com/)

**Comment by AyaanZaveri** [comment]
The only one that worked for me (other than Heroku and Okteto) is Render, but I'm on a 7-day trial, so I don't think it will work after the trial is over.

**Comment by omgbox** [comment]
KiraLT you don't have to use free trial if you do small amount of survey work from [prizerebel](https://www.prizerebel.com/index.php?r=11648173) 1000 points is $10 usd you could make this easily in few days on and off to pay for a a medium ovh or hetzner server, took me about nearly 18 surveys to get to 1000 points $10USD, after getting the points you can send to your paypal or visa virtual card and use it for anything.  trust me guys it is worth the time. just signup and do few surveys, i know its not alot but you could easily pay every month your vps this way. and only takes like 2 hours per week . don't click on the surveys fast or you face getting blocked by some of the survey sites. go[ here](https://www.prizerebel.com/index.php?r=11648173&usub=github) and signup if you want to pay for your VPS server in few weeks. 

i worked few hours only per week. i have already nearly made $30USD and don't have to worry about VPS 
![Capture](https://user-images.githubusercontent.com/11356775/124338137-1fd06980-db9e-11eb-8f01-40448bed3586.JPG)
from paypal history
![2](https://user-images.githubusercontent.com/11356775/124338138-2363f080-db9e-11eb-9a5d-0004fb2456c4.JPG)

i hope this help you guys, if you have some questions let me know on here or on my github. take care

**Comment by AyaanZaveri** [comment]
What about Cloudflare workers?

**Comment by omgbox** [comment]
> What about Cloudflare workers?

so weird, i was just looking at that on cloudflare 

> Cloudflare Workers is a serverless application platform that enables you to create and deploy functions to Cloudflare's Edge Network. Unlike other serverless providers that only have regional data centers, Cloudflare's Edge Network consists of hundreds of servers all over the world.

might be possible i have no idea, it essentially spreads your javascript functions on 100s of servers. have no idea how they would deal with storage , cloudflare has free version,  even the paid account is $5usd very cheap. 

more info [here](https://blog.cloudflare.com/workers-kv-free-tier/)

**Comment by AyaanZaveri** [comment]
Did you find anything?

**Comment by omgbox** [comment]
> Did you find anything?

nothing yet,  but found this interesting [https://github.com/niutech/node.php](url) run node inside php, many free hosting providers offer php but many of the important modules are turned off . but still very interesting .

**Comment by AyaanZaveri** [comment]
Pretty cool!

**Comment by AyaanZaveri** [comment]
I'm surprised there are only 2 services (Heroku and Okteto) that work and are free!

**Comment by pataquets** [comment]
Check out Google Cloud Platform and Amazon Web Services, both offer a free tier. Watch out outgoing transfer, mainly.

Yesterday I was looking for something like a "one click deploy" as in Heroku I remembered seeing in GCP, but did not found it. I might dig a bit deeper if I need it in the future, just letting you know.

**Comment by omgbox** [comment]
> Check out Google Cloud Platform and Amazon Web Services, both offer a free tier. Watch out outgoing transfer, mainly.
> 
> Yesterday I was looking for something like a "one click deploy" as in Heroku I remembered seeing in GCP, but did not found it. I might dig a bit deeper if I need it in the future, just letting you know.

GCP and AWS i think both need you to add your credit card , if you pass the free limits you will be automatically charged. AWS limit outgoing bandwidth is only 5gb i believe and same with GCP. they good if you can pay the rent. today found another interesting cloud based container deployment provider [here](https://akash.network/) i believe for now they offer dev free tokens which can be used to try the service.

**Comment by pataquets** [comment]
As for GCP stands, that's not exact:
* Adding credit card is to verify you're a human and prevent bots.
* After the free trial ($300), you have to *explicitly* upgrade to a paid
plan (which, indeed, includes a free tier).
See https://cloud.google.com/free/docs/gcp-free-tier for
more information. Anyway, people are supposed to do their homework and check if it suits them before choosing.

Taking into account that now TSS has bandwidth limits, it might fit more use cases in scenarios where you have quotas. BTW, I didn't know Akash, will look at it. Thanks for sharing!

Also, I'm wondering: since free services and tiers come and go, will we be OK on keeping the list up to date or would it be better to just provide a link to free-for.dev website? Just a thought.

**Comment by omgbox** [comment]
right now I am trying to use openode.io 100gb per month which is free so is enough for lots of anime :), there is also gigalixir which I tried it is very complicated and failed . so right now I'm fighting with starting up Torrent-stream-server on openode.io which is giving me  "status": "failed", . I have even changed the Docker File so not sure what the issue is. wish me goo luck U_U

**Comment by AyaanZaveri** [comment]
Openode.io failed for me too, I don't know how to deploy on there.

**Comment by omgbox** [comment]
This Comment has been deleted By 
**United States Government in Accordance with Section 17c Part 9c Law,**  LOL 

yeah its done

**Comment by KiraLT** [comment]
I am exploring deployment on [Deta](https://www.deta.sh/) (it also supports one-click deployment like Heroku). You can follow the progress on #247.

**Comment by omgbox** [comment]
found another one, [fly.io](https://fly.io/docs/about/pricing/) provide heroku style free hosting 

<html>
<body>
<h2 id="free-allowances" class="group flex items-center whitespace-pre-wrap relative mt-14 sm:mt-16 mb-10 text-navy font-heading pb-1 border-b border-gray-100" style="box-sizing: border-box; border-width: 0px 0px 1px; border-style: solid; border-color: rgb(241 242 249/var(--tw-border-opacity)); border-image: initial; --tw-translate-x:0; --tw-translate-y:0; --tw-rotate:0; --tw-skew-x:0; --tw-skew-y:0; --tw-scale-x:1; --tw-scale-y:1; --tw-pan-x: ; --tw-pan-y: ; --tw-pinch-zoom: ; --tw-scroll-snap-strictness:proximity; --tw-ordinal: ; --tw-slashed-zero: ; --tw-numeric-figure: ; --tw-numeric-spacing: ; --tw-numeric-fraction: ; --tw-ring-inset: ; --tw-ring-offset-width:0px; --tw-ring-offset-color:#fff; --tw-ring-color:rgba(59,130,246,0.5); --tw-ring-offset-shadow:0 0 transparent; --tw-ring-shadow:0 0 transparent; --tw-shadow:0 0 transparent; --tw-shadow-colored:0 0 transparent; --tw-blur: ; --tw-brightness: ; --tw-contrast: ; --tw-grayscale: ; --tw-hue-rotate: ; --tw-invert: ; --tw-saturate: ; --tw-sepia: ; --tw-drop-shadow: ; --tw-backdrop-blur: ; --tw-backdrop-brightness: ; --tw-backdrop-contrast: ; --tw-backdrop-grayscale: ; --tw-backdrop-hue-rotate: ; --tw-backdrop-invert: ; --tw-backdrop-opacity: ; --tw-backdrop-saturate: ; --tw-backdrop-sepia: ; font-size: 1.5rem; font-weight: 700; margin: 2.75rem 0px 2.5rem; font-family: Mackinac, ui-serif, Georgia, Cambria, &quot;Times New Roman&quot;, Times, serif; letter-spacing: -0.025em; word-spacing: 0.025em; position: relative; display: flex; align-items: center; white-space: pre-wrap; --tw-border-opacity:1; padding-bottom: 0.25rem; --tw-text-opacity:1; color: rgb(36 24 91/var(--tw-text-opacity)); line-height: 2rem; scroll-margin-top: 9rem; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Free Allowances</h2><div class="min-w-0 overflow-x-auto bg-white shadow-md rounded-xl lg:-mx-8 mb-7 ring-1 ring-gray-700 ring-opacity-10" style="box-sizing: border-box; border: 0px solid rgb(225, 228, 239); --tw-translate-x:0; --tw-translate-y:0; --tw-rotate:0; --tw-skew-x:0; --tw-skew-y:0; --tw-scale-x:1; --tw-scale-y:1; --tw-pan-x: ; --tw-pan-y: ; --tw-pinch-zoom: ; --tw-scroll-snap-strictness:proximity; --tw-ordinal: ; --tw-slashed-zero: ; --tw-numeric-figure: ; --tw-numeric-spacing: ; --tw-numeric-fraction: ; --tw-ring-inset: ; --tw-ring-offset-width:0px; --tw-ring-offset-color:#fff; --tw-ring-color:rgb(55 57 83/var(--tw-ring-opacity)); --tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color); --tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color); --tw-shadow:0 4px 6px -1px rgba(0,0,0,0.1),0 2px 4px -2px rgba(0,0,0,0.1); --tw-shadow-colored:0 4px 6px -1px var(--tw-shadow-color),0 2px 4px -2px var(--tw-shadow-color); --tw-blur: ; --tw-brightness: ; --tw-contrast: ; --tw-grayscale: ; --tw-hue-rotate: ; --tw-invert: ; --tw-saturate: ; --tw-sepia: ; --tw-drop-shadow: ; --tw-backdrop-blur: ; --tw-backdrop-brightness: ; --tw-backdrop-contrast: ; --tw-backdrop-grayscale: ; --tw-backdrop-hue-rotate: ; --tw-backdrop-invert: ; --tw-backdrop-opacity: ; --tw-backdrop-saturate: ; --tw-backdrop-sepia: ; margin-bottom: 1.75rem; min-width: 0px; overflow-x: auto; border-radius: 0.75rem; --tw-bg-opacity:1; background-color: rgb(255 255 255/var(--tw-bg-opacity)); box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 transparent); --tw-ring-opacity:0.1; margin-left: 1px; margin-right: 1px; color: rgb(76, 78, 103); font-family: Inter, ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

VM: shared-cpu | 2,340 hours per month | Run 3 shared-cpu-1x VMs with 256MB RAM full time.
-- | -- | --
3GB | Provision 3GB of persistent volumes for permanent storage
160GB per month | See outbound data transfer for regional breakdown
Unlimited IPv6, 1 IPv4 per active app | Additional IPv4 addresses are $2 per month
10 active certificates | Add 10 certificates to your apps

**Comment by omgbox** [comment]
found another one, called [deno ](https://deno.com/deploy/pricing)

Deploy JavaScript Globally
Instant deployments
Located in [33 regions](https://deno.com/deploy/docs/regions) worldwide
Zero config, zero maintenance
TypeScript, Wasm, ES Modules

$0 /mo

100,000 requests per day
100 GiB data transfer per month
